front/src/applications/operationalStudies/views/Scenario/components/ScenarioContent.tsx

```diff
@@ -56,6 +58,30 @@ const ScenarioContent = ({ activeBoards }: ScenarioContentProps) => {
   const [collapsedTimetable, setCollapsedTimetable] = useState(false);
   const [collapsedTimetableEdit, setCollapsedTimetableEdit] = useState(false);
   const [timetableItemToEditData, setTimetableItemToEditData] = useState<TimetableItemToEditData>();
+  const [macroBoardHeight, setMacroBoardHeight] = useState<number>(MACRO_EDITOR_HEIGHT);
+  const macroBoardRef = useRef<HTMLDivElement>(null);
+
+  // Calculate default height to take full available space
+  const getDefaultMacroBoardHeight = useCallback(() => {
+    if (macroBoardRef.current) {
+      const centerColumn = macroBoardRef.current.closest('.center-column');
+      if (centerColumn) {
+        const centerColumnHeight = centerColumn.clientHeight;
+        const padding = 60;
+        const headerHeight = 32;
+        return centerColumnHeight - padding - headerHeight;
+      }
+    }
+    return MACRO_EDITOR_HEIGHT;
+  }, []);
+
+  // Set initial height to full available space when macro is active
+  useEffect(() => {
+    if (activeBoards.has('macro')) {
+      const defaultHeight = getDefaultMacroBoardHeight();
+      setMacroBoardHeight(defaultHeight);
+    }
+  }, [activeBoards, getDefaultMacroBoardHeight]);
```

Comment:
Since getDefaultMacroBoardHeight is only used inside the effect, I think you don’t need to memoize it:

```suggestion
  useEffect(() => {
  if (activeBoards.has('macro') && macroBoardRef.current) {
    const centerColumn = macroBoardRef.current.closest('.center-column');
    const centerColumnHeight = centerColumn?.clientHeight ?? 0;
    const padding = 60;
    const headerHeight = 32;

    setMacroBoardHeight(
      centerColumnHeight
        ? centerColumnHeight - padding - headerHeight
        : MACRO_EDITOR_HEIGHT
    );
  }
}, [activeBoards]);
```